### PR TITLE
Tests for corner cases in MP FT integration with Micrometer and MP Telemetry

### LINKED ISSUE #317: Add tests to verify corner cases in MP FT integration with metrics generation subsystems, i.e. Micrometer and MP Telemetry
During the implementation of the upgrade tracked by https://issues.redhat.com/browse/WFLY-19592, a new functionality has been added to SmallRye Fault Tolerance, that affects the WIldFly integration as well, i.e. https://github.com/smallrye/smallrye-fault-tolerance/issues/1078

Such functionality is good since it allows users to tweak the configuration in case multiple metrics providers are _available_ and/or _enabled_.
Tests covering such functionality exist in the SmallRye Fault Tolerance test suite, but we've identified some combinations that should be tested at the WildFly level in order to validate the integration and define the correct server behavior, i.e. based on the end user (DevOps, Admins) PoV, that is to say, _when related **extensions** and **subsystems** are **available or enabed**_, and to keep regressions under control.

The following combinations were identified:

1. no metrics providers available
2. only Micrometer available and enabled
3. only MP Telemetry available and enabled
4. both Micrometer and MP Telemetry extensions available and enabled
5. both Micrometer and MP Telemetry extensions available, but MP Telemetry is disabled
6. both Micrometer and MP Telemetry extensions available, but Micrometer is disabled
7. both Micrometer and MP Telemetry extensions available, but both are disabled

Tests for 4, 5, 7 are provided already by the should be WildFly test suite integration tests.

This issue tracks the effort needed to add tests for the remaining cases (1, 2, 3 and 6).

### PR DESCRIPTION
**Pull request by fabiobrz**
SSIA

Resolves #317

**Validation job runs**:

- Manual workflow run (bootable  JAR execution): ~https://github.com/fabiobrz/eap-microprofile-test-suite/actions/runs/12135061804~ ~https://github.com/fabiobrz/eap-microprofile-test-suite/actions/runs/12142399569~ https://github.com/jboss-eap-qe/eap-microprofile-test-suite/actions/runs/12142663575 - _1 failure in `DatabaseCrashTest` is unrelated_
- Manual workflow run (regular execution): ~https://github.com/fabiobrz/eap-microprofile-test-suite/actions/runs/12135055664~ ~https://github.com/fabiobrz/eap-microprofile-test-suite/actions/runs/12142389442~ https://github.com/jboss-eap-qe/eap-microprofile-test-suite/actions/runs/12142663573 - _1 failure in `DatabaseCrashTest` is unrelated_

---
Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Link to the passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

### CODE REVIEW

File: microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+        String mpConfig = "otel.service.name=" + FAULT_TOLERANCE_DEPOYMENT_WITH_MP_TELEMETRY_DISABLING_MICROMETER
+                + "\notel.sdk.disabled=false\nTimeout/enabled=true\nsmallrye.faulttolerance.micrometer.disabled=true";
+        return ShrinkWrap.create(WebArchive.class, FAULT_TOLERANCE_DEPOYMENT_WITH_MP_TELEMETRY_DISABLING_MICROMETER + ".war")
+                .addPackages(true, HelloService.class.getPackage())
+                .addAsManifestResource(ConfigurationUtil.BEANS_XML_FILE_LOCATION, "beans.xml")
+                .addAsManifestResource(new StringAsset(mpConfig), "microprofile-config.properties");
+    }
+
+    /**
+     * Deploy all deployments, so it's possible to get their URLs in other tests. This is limitation of Arquillian
```
Comment (marekkopecky) [inline microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java:120]: Do we need to inject URL by arquillian? Host name and port are hard-coded already in many places around the TS and URL is http://127.0.0.1:8080/deployment-name. This workaround seems to be expensive ...

File: microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+            try {
+                // call the app operation for some times
+                for (int i = 0; i < REQUEST_COUNT; i++) {
+                    get(deploymentUrl + "?operation=timeout&context=foobar&fail=true").then()
+                            .assertThat()
+                            .statusCode(HttpStatus.SC_OK)
+                            .body(containsString("Fallback Hello, context = foobar"));
+                }
+                // give it some time to actually be able and report some metrics via the Pmetheus URL
+                Thread.sleep(5_000);
```
Comment (marekkopecky) [inline microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java:167]: Can we decrease this a little bit by adding "otel.metric.export.interval=100" to mp config property file?

File: tooling-observability/src/main/java/org/jboss/eap/qe/observability/containers/OpenTelemetryCollectorContainer.java [hunk 3]
```diff
@@ -103,16 +110,38 @@ private OpenTelemetryCollectorContainer() {
...
     public static synchronized OpenTelemetryCollectorContainer getInstance() {
         if (INSTANCE == null) {
-            jaegerContainer = JaegerContainer.getInstance();
             INSTANCE = new OpenTelemetryCollectorContainer();
-            INSTANCE.start();
         }
         return INSTANCE;
     }
 
+    public static synchronized OpenTelemetryCollectorContainer getInstance(JaegerContainer jaegerBackendContainer) {
```
Comment (marekkopecky) [inline tooling-observability/src/main/java/org/jboss/eap/qe/observability/containers/OpenTelemetryCollectorContainer.java:118]: Can you please add javadoc to new public methods?

**Comment by marekkopecky** [comment]
Just a few minor comments, otherwise LGTM

**Comment by marekkopecky** [comment]
Btw, let's merge this PR, then I will rebase https://github.com/jboss-eap-qe/eap-microprofile-test-suite/pull/315 and we can continue with processing of https://github.com/jboss-eap-qe/eap-microprofile-test-suite/pull/315

File: microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java (hunk 1 above)
Comment (fabiobrz) [reply microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java:120]: We do need it. We can't allow for multiple deployments to be there at the same time, otherwise we would have un-deterministic behavior with respect to metrics collection.

The deployAll() method is something that we have to do if we want to deploy manually (required here) and be able to inject the URLs in the tests methods as Arquillian resources,

File: microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java (hunk 2 above)
Comment (fabiobrz) [reply microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java:167]: Thanks for your suggestion, experimenting with it now.

File: tooling-observability/src/main/java/org/jboss/eap/qe/observability/containers/OpenTelemetryCollectorContainer.java (hunk 3 above)
Comment (fabiobrz) [reply tooling-observability/src/main/java/org/jboss/eap/qe/observability/containers/OpenTelemetryCollectorContainer.java:118]: +1, sure! Fixing...

**Comment by fabiobrz** [comment]
> Btw, let's merge this PR, then I will rebase #315 and we can continue with processing of #315

Hi @marekkopecky - I've addressed al of your coments above, and validation runs are ok, being the `DatabaseCrashTest` failure unrelated.

File: microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java (hunk 1 above)
Comment (marekkopecky) [reply microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java:120]: Let's approve it now, I will double-check this next week and maybe create some follow-up issue ...

**Comment by marekkopecky** [comment]
Thank you for the discussion here, all threads are resolved, merging.

**Review by marekkopecky — APPROVED** [review]
